syntax = "proto3";

package emb;

option go_package = "./emb";

service UserEmbeddingService {
  rpc CreateUserEmbedding(UserEmbeddingRequest) returns (EmbeddingResponse);
  rpc UpdateUserEmbedding(UserEmbeddingRequest) returns (EmbeddingResponse);
  rpc UpdateUserStatus(UpdateStatusRequest) returns (EmbeddingResponse);
  rpc DeleteUserEmbedding(DeleteEmbeddingRequest) returns (EmbeddingResponse);
}

service ProjectEmbeddingService {
  rpc CreateProjectEmbedding(ProjectEmbeddingRequest) returns (EmbeddingResponse);
  rpc UpdateProjectEmbedding(ProjectEmbeddingRequest) returns (EmbeddingResponse);
  rpc UpdateProjectStatus(UpdateStatusRequest) returns (EmbeddingResponse);
  rpc DeleteProjectEmbedding(DeleteEmbeddingRequest) returns (EmbeddingResponse);
}

message UserEmbeddingRequest {
  string user_id = 1;
  string bio = 2;
  repeated string skills = 3;
  repeated string interests = 4;
}

message ProjectEmbeddingRequest {
  string project_id = 1;
  string title = 2;
  string description = 3;
  repeated string skills = 4;
}

message UpdateStatusRequest {
  string id = 1;
  bool status = 2;
}

message DeleteEmbeddingRequest {
  string id = 1;
}

message EmbeddingResponse {
  bool success = 1;
  string msg = 2;
}
